log/noop_logger.go

```diff
@@ -6,14 +6,14 @@ var _ Logger = &NoOp{}
 
 type NoOp struct{}
 
-func NewNoOpLogger() *NoOp {
+func NewNopLogger() Logger {
 	return &NoOp{}
 }
 
-func (l NoOp) Debug(msg string, keyvals ...interface{}) {}
-func (l NoOp) Info(msg string, keyvals ...interface{})  {}
-func (l NoOp) Error(msg string, keyvals ...interface{}) {}
+func (l *NoOp) Debug(msg string, keyvals ...interface{}) {}
```

Comment:
I think we can implement this without a pointer receiver considering the size of a `struct{}` in go is 0.

Reply:
Oh will change. Thanks for the insight!